types/types.go

```diff
@@ -355,3 +355,34 @@ func (jm JSONMap) GormValue(ctx context.Context, db *gorm.DB) clause.Expr {
 	data, _ := jm.MarshalJSON()
 	return gorm.Expr("?", string(data))
 }
+
+// GenericStructValue can be set as the Value() func for any json struct
+func GenericStructValue[T any](t T, defaultNull bool) (driver.Value, error) {
+	b, err := json.Marshal(t)
+	if err != nil {
+		return b, err
+	}
+	if defaultNull && string(b) == "{}" {
+		return nil, nil
+	}
+	return string(b), nil
+}
+
+// GenericStructScan can be set as the Scan(val) func for any json struct
+func GenericStructScan[T any](t *T, val any) error {
+	if val == nil {
+		*t = *new(T)
```

Comment:
If `t` is a nil pointer then this panics. But the way we're using this func guarantees what it won't ever be nil. So I guess it's fine.

Although this could just be `t = new(T)`.

Reply:
👍🏽